render() got an unexpected keyword argument 'renderer'

jikoy opened the pull request:
Request Method: | GET
-- | --
http://127.0.0.1:8000/admin/blog/article/1/change/
2.1.4
TypeError
render() got an unexpected keyword argument 'renderer'
F:\mysite\UEditor\venv\lib\site-packages\django\forms\boundfield.py in as_widget, line 93
F:\mysite\UEditor\venv\Scripts\python.exe
3.7.1

解决方法，不知道是否正确。我注释了 boundfield.py 第93行，就能运行了。大牛看一下。

twz915 committed:
兼容Python3改进

Commit:
Revert "兼容Python3改进"

This reverts commit 39742666ce8ea9987be7f062f71e0a802f8ee27a.

Commit:
兼容python3 修改

已在 python3.4 + django1.8 上测试过，可以正常使用，只是上传图片一直失败。

Revert "Revert "兼容Python3改进""

This reverts commit
f612eeb776320a73b5cb7d40de64e7e323fbde27.

Revert "兼容Python3改进"

This reverts commit
39742666ce8ea9987be7f062f71e0a802f8ee27a.

twz915 committed:
Merge pull request #1 from xinyvz/master

Master

Commit:
remove *.pyc and add .gitignore file

Commit:
合并历史修改

Commit:
兼容 Python3

sazima commented:
提一句：DjangoUeditor出现bug，设法去修改django的源码是非常不明智的。 


解决方法： 将DjangoUeditor中的render方式加上一些可选参数就好了： 

将：
```
 def render(self, name, value, attrs=None):
```
改为： 
```
  def render(self, name, value, attrs=None, *args, **kwargs):
```

echo1937 committed:
fix widgets.py for Django >= 2.1:
def render(self, name, value, attrs=None):
# 修改为
def render(self, name, value, attrs=None, renderer=None):

twz915 committed:
Merge pull request #2 from echo1937/master

fix widgets.py for Django >= 2.1:

twz915 committed:
fix json response

twz915 committed:
adapt latest Django version

twz915 committed:
fix setup

twz915 committed:
remove .idea
